src/CastleAdventure.java

```diff
@@ -5,27 +5,87 @@ public class CastleAdventure {
     public static void main(String[] args) {
         Scanner scanner = new Scanner(System.in);
 
-        // Нижче перша частина завдання для прикладу
-        // Там де в завданні вказані початкові дані у вигляді масиву - ініціалізуйте окремий масив
-        // Нижче обробляйте масив, додайте логіку обробки
-        // Частина 1: Увійди в Замок
         int[] secretCode = {7, 14, 28, 56, 112};
         System.out.println("Введіть 5 чисел, щоб відкрити двері замку:");
         boolean isCodeCorrect = true;
 
         for (int i = 0; i < secretCode.length; i++) {
-            int userInput = scanner.nextInt(); // питаємо в користувача чергове число
-            if (userInput != secretCode[i]) { // перевіряємо число проти i-го елемента масиву з правильним кодом
-                isCodeCorrect = false; // якщо натрапляємо на неправильно введене число - усю комбінацію вважаємо невірною
+            System.out.print(" Введи число " + (i + 1) + " : ");
+            int userInput = scanner.nextInt();
+            if (userInput != secretCode[i]) {
+                isCodeCorrect = false;
             }
         }
 
         if (isCodeCorrect) {
             System.out.println("Замок відкрито! Заходьте всередину.");
         } else {
             System.out.println("Невірний код. Спробуйте ще.");
+            scanner.close();
+            return;
         }
 
-        // Інші частини пригоди додайте тут!
+        System.out.println("Потрапивши у замок, ти бачиш чарівну дошку з числами.Розв'яжи загадку");
+
+        int[] magicBoard = {3, 6, 9, 12, 15, 18, 21, 24, 27, 30};
+        System.out.print("На дошці написано числа: ");
+        for (int number : magicBoard) {
+            System.out.print(number + " ");
+        }
+        System.out.println();
+
+        int actualSum = 0;
+        for (int number : magicBoard) {
+            actualSum += number;
+        }
+
+        System.out.println("Сума всіх магічних чисел: " );
+        int userSum = scanner.nextInt();
+        if (userSum == actualSum) {
+        System.out.println("Вітаю!Кімната відкрилась!Ти можеш йти далі.");
+        } else {
+            System.out.println("Невірно.Спробуй ще.");
+        }
+        System.out.println("Перед тобою з'явився Дракон!Захищайся,активувуй захист!");
+
+        int[] dragonAttacks = {8, 16, 24, 32, 40};
+        boolean allEven = true;
+
+        for (int attack : dragonAttacks) {
+            if (attack % 2 != 0) {
+                allEven = false;
+                break;
+            }
+        }
+        if (allEven) {
+            System.out.println("Атаки парні! Захист активовано.");
+        } else {
+            System.out.println("Увага!Непарна атака!");
+        }
+
+        System.out.println("Щоб знайти магічний ключ,введи своє улюблене число:");
+        int favoriteNumber = scanner.nextInt();
+        boolean found = false;
+
+        int[][] allSources = { secretCode, magicBoard, dragonAttacks };
+
+        outerLoop:
```

Comment:
A loop label!
I haven't seen it for years

Here I'd say it's understandable why you use it (because at that moment we didn't have enough instruments to make it gracefully) - although that's a pretty interesting solution.


Btw just an interesting part of history - this functionality was added to Java since Java 1.0 in 1995, now almost nobody uses it so you found a hidden gem :) (although probably this will be the only time you use it ever hehee)